src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/PersonRepository.kt

```diff
@@ -1,8 +1,46 @@
 package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository
 
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
 import org.springframework.data.repository.CrudRepository
 import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Person
+import java.time.LocalDateTime
 
 @Repository
-interface PersonRepository : CrudRepository<Person, Long>
+interface PersonRepository :
+  CrudRepository<Person, Long>,
+  JpaRepository<Person, Long> {
+  @Query(
+    """
+      select 
+        p.id as personId
+      from Person p 
+    """,
+  )
+  fun findAllPersonIds(
+    pageable: Pageable,
+  ): Page<PersonIdProjection>
+
+  @Query(
+    """
+      select 
+        p.id as personId
+      from Person p 
+        where 
+          (:fromDate is null or p.createDatetime > :fromDate) and 
```

Comment:
this won't match people with a from date at the beginning of the day i.e. should it be `>=`?

Reply:
The service adjusts the date (which are actually dates ) to make this work